static/app/components/replays/replayContext.tsx

```diff
@@ -229,6 +241,11 @@ export function Provider({children, replay, initialTimeOffset = 0, value = {}}:
     removeHighlightedNode({replayer, nodeId});
   }, []);
 
+  const setReplayFinished = useCallback(() => {
+    setFinishedAtMS(replayerRef.current?.getCurrentTime() ?? -1);
+    setIsPlaying(false);
+  }, [replayerRef]);
```

Comment:
Good catch I never thought about this before so use to just adding deps to the array. I think in this case I actually want to track replayerRef.current to be technically correct since if ref.current isn't set initially the effect/callback will never be run again. If replayerRef.current gets mutated it won't cause a re-render, but if one does happen and it's value has changed, the effect will be run.

Just made a StackBlitz to test this out: https://stackblitz.com/edit/react-ts-nh5ace?file=App.tsx